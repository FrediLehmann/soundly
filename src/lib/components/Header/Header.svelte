<nav class="bg-gray-800">
  <div class="max-w-7xl mx-auto px-2 sm:px-6 lg:px-8 flex">
    <img src="/static/favicon/favicon-32x32.png" alt="Logo" />
    <span>Soundly</span>
  </div>
</nav>
