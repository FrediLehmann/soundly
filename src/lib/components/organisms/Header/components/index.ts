import IconSection from './IconSection.svelte';
import AccountSection from './AccountSection.svelte';

export { IconSection, AccountSection };
