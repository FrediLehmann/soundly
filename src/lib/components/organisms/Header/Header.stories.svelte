<script>
  import { Meta, Story } from '@storybook/addon-svelte-csf';
  import { userEvent, within } from '@storybook/testing-library';

  import { Navigation, AccountSection } from './components';
  import Header from './Header.svelte';
</script>

<Meta title="Organisms/Header" />

<Story
  name="Navigation part"
  play={async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const link = canvas.getAllByRole('link')[0];

    userEvent.hover(link);
  }}>
  <div class="navcontainer">
    <Navigation />
  </div>
</Story>

<Story name="Account part">
  <AccountSection />
</Story>

<Story name="Full header">
  <Header />
</Story>

<style>
  .navcontainer {
    display: flex;
  }
</style>
