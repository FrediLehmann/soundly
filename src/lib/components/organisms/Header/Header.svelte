<script lang="ts" context="module">
  import { Logo } from '$lib/components/atoms';

  import { AccountSection, Navigation } from './components';
</script>

<header>
  <Logo href="/" size="medium" />
  <Navigation />
  <AccountSection />
</header>

<style>
  header {
    display: flex;
    margin-inline: auto;
    padding-block: 1rem;
    padding-inline: 0.5rem;
    max-width: 72rem;
    align-items: center;
  }
</style>
