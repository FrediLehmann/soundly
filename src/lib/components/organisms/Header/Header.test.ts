/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import Header from './Header.svelte';

describe('<Header /> component', () => {
  test('renders correctly', () => {
    const { component } = render(Header);
    expect(component).toMatchSnapshot();
  });
});
