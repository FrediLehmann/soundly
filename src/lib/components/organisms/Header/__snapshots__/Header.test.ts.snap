// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component renders correctly 1`] = `
Header {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {},
    "root": <div>
      <header
        class="svelte-wrs6mn"
      >
        <picture
          class="svelte-1svwpk4"
          data-size="medium"
        >
          <source />
           
          <img
            alt="Noiceless Logo"
            class="svelte-1svwpk4"
          />
        </picture>
         
        <nav
          class="svelte-1upcm0y"
        >
          <ul
            class="svelte-1upcm0y"
          >
            <li
              class="svelte-1upcm0y"
            >
              <a
                class="svelte-1upcm0y"
                href="/upload"
                sveltekit:prefetch=""
              >
                Upload
              </a>
            </li>
             
            <li
              class="svelte-1upcm0y"
            >
              <a
                class="svelte-1upcm0y"
                href="/about"
                sveltekit:prefetch=""
              >
                About
              </a>
            </li>
             
            <li
              class="svelte-1upcm0y"
            >
              <a
                class="svelte-1upcm0y"
                href="/api"
                sveltekit:prefetch=""
              >
                API
              </a>
            </li>
          </ul>
        </nav>
         
        <div
          class="svelte-8vfevs"
        >
          <a
            aria-label="Profile"
            class="svelte-1dsjpe6"
            data-type="ghost"
            href="/profile"
            type="button"
          >
            <svg
              fill="none"
              height="16"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
              />
              <circle
                cx="12"
                cy="7"
                r="4"
              />
            </svg>
          </a>
          
           
          <a
            class="svelte-1dsjpe6"
            data-type="ghost"
            href="/signin"
            type="button"
          >
            Sign in
          </a>
          
        </div>
      </header>
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$capture_state": [Function],
}
`;
